packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -493,8 +537,12 @@ class FlutterPlugin implements Plugin<Project> {
             String maxPluginNdkVersion = projectNdkVersion
             int numProcessedPlugins = getPluginList().size()
 
-            getPluginList().each { plugin ->
-                Project pluginProject = project.rootProject.findProject(plugin.key)
+            getPluginList().each { pluginObject ->
+                assert pluginObject.name instanceof String
+                Project pluginProject = project.rootProject.findProject(":${pluginObject.name}")
+                if (pluginProject == null) {
```

Comment:
Can you add a comment explaining why if a plugin does not have a name why returning is the correct behavior (as opposed to an error or logged warning)

Reply:
Good comments in new code explaining this behavior.